updater/index.php

```diff
@@ -126,10 +126,14 @@ function get_array_by_delimiter($subject, $delimiter)
 	if (!Utility::is_extension_in_list($_FILES['patchfile']['name'], $allowed_file_extensions)) {
 		$msg->addError(array('ALLOWED_FILE_TYPES', implode(", ", $allowed_file_extensions)));
 	} else {
-		$archive = new PclZip($_FILES['patchfile']['tmp_name']);
+		$zip = new ZipArchive();
 
-		if ($archive->extract(PCLZIP_OPT_PATH, $module_content_folder) == 0)
-		{
+		$zip->open($_FILES['patchfile']['tmp_name']);
+
+		if ($zip->open($_FILES['patchfile']['tmp_name'])){

```

Comment:
Redundant same call to `open()` at line 131?

Reply:
This line should also ensure the returned `TRUE` from the open() call. Otherwise, if a non-zero error code is returned, the if block would still accept it and run. Also, use [`===`](http://us3.php.net/manual/en/language.operators.comparison.php) to verify `TRUE` value.